tarentula/datashare_client.py

```diff
@@ -194,3 +194,25 @@ def temporary_project(self, source=DATASHARE_DEFAULT_PROJECT, delete=True):
             if delete and project is not None:
                 self.delete_index(project)
         return project
+
+    def metadata_fields(self, index=DATASHARE_DEFAULT_PROJECT, document_type='Document'):
+        url = urljoin(self.elasticsearch_host, index)
+        mapping = requests.get(url, cookies=self.cookies, headers=self.headers).json()
+        results = []
+        self.get_fields(document_type, index, mapping, results, [])
```

Comment:
We should be able to return the results of this method directly (see my next comment about immutable).